PWGGA/GammaConv/AliConvEventCuts.cxx

```diff
@@ -4847,23 +4848,11 @@ Float_t AliConvEventCuts::GetWeightForMeson(Int_t index, AliMCEvent *mcEvent, Al
     functionResultData = fFitDataK0s->Eval(mesonPt);
   }
 
+  // calculate weight from data and MC
   Double_t weight = 1;
   if (PDGCode ==  111 || PDGCode ==  221){
     if (functionResultData != 0. && functionResultMC != 0. && isfinite(functionResultData) && isfinite(functionResultMC)){
       weight = functionResultData/functionResultMC;
-      if ( kCaseGen == 3){   // never true ?
-        if (PDGCode ==  111){
-        if (!(fDoReweightHistoMCPi0 && hReweightMCHistPi0!= 0x0 && PDGCode ==  111)){
-          weight = 1.;
-        }
-        }
-        if (PDGCode ==  221){
-        if (!(fDoReweightHistoMCEta && hReweightMCHistEta!= 0x0 && PDGCode ==  221)){
-          weight = 1.;
-        }
-        }
-      }
-      if (!isfinite(functionResultData)) weight = 1.;
```

Comment:
ok but how can kCaseGen == 3 ever be true? It is set to 0 when the function is entered and then within the function it can be set to 1 (or 2) or otherwise stay 0